### ISSUE TITLE
Token-storing off-server magic vault thing

### ISSUE THREAD

**Issue by Undo1** [Charcoal-SE/metasmoke#529]
Details to come. @Undo1 write some stuff here in the morning.

**Comment by ArtOfCode-** [comment]
I've been vaguely thinking about this. The root of the problem here is that any such system will rely on having a data store/environment that you can insert to, but that you cannot retrieve from. I can't think of any of those off the top of my head.

**Comment by thesecretmaster** [comment]
The idea was a lambda + dynamodb IIRC

**Comment by Undo1** [comment]
Whoever runs it will probably be able to access them, but the benefit from moving tokens over would be a much-reduced attack surface. It'd be only a few web routes, no DB access from metasmoke (so no attack surface for database dumps), no automatic deploys, etc. No way for even MS deployers to ever get access, so definitely no way for attackers to get it. 

tl;dr - `u.api_token` is pretty easy for anyone with even a bit of access to run. This wouldn't be.

**Comment by ArtOfCode-** [comment]
If whoever's hosting will probably still be able to access them, that makes it significantly easier. Two options come to mind:

 - Lambda + DynamoDB
 - YAUWS (Yet Another Unstable Web Service)

Guess which one's probably the sensible option :)

**Comment by thesecretmaster** [comment]
I've built a little thing for this. It's got three routes:

1. /auth, does SE oauth and stores token
2. /auth/complete, also a part of SE oauth
3. /autoflag, keyed, post, MS uses this to cast flags with params for account_id, comment, flag_option_id, post_id, post_type, site

I figured MS could get flag_option_id with Smokey's account. If not, doing flag option as a string wouldn't be too horrible, I don't think.
